### PR TITLE
etcdserver: Extract functions for setting and reading compaction information in backend

### PR DESCRIPTION
**Pull request by serathius**
cc @lilic @ptabor

### CODE REVIEW

File: server/mvcc/kvstore.go [hunk 1]
```diff
@@ -339,10 +333,10 @@ func (s *store) restore() error {
 	tx := s.b.BatchTx()
 	tx.Lock()
 
-	_, finishedCompactBytes := tx.UnsafeRange(buckets.Meta, buckets.FinishedCompactKeyName, nil, 0)
-	if len(finishedCompactBytes) != 0 {
+	finishedCompact, found := UnsafeReadFinishedmpact(tx)
```
Comment (ptabor) [inline server/mvcc/kvstore.go:336]: UnsafeReadFinishedCompat(tx) ?

File: server/mvcc/store.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// limitations under the License.
+
+package mvcc
+
+import (
+	"go.etcd.io/etcd/server/v3/mvcc/backend"
+	"go.etcd.io/etcd/server/v3/mvcc/buckets"
+)
+
+func UnsafeReadFinishedmpact(tx backend.ReadTx) (finishedComact int64, found bool) {
```
Comment (ptabor) [inline server/mvcc/store.go:22]: UnsafeReadFinishedCompat

File: server/mvcc/store_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			b := backend.NewDefaultBackend(tmpPath)
+			defer b.Close()
+			v, found := UnsafeReadScheduledCompact(b.BatchTx())
+			assert.Equal(t, true, found)
+			assert.Equal(t, tc.value, v)
+		})
+	}
+}
+
+// TestFinishedCompact ensures that UnsafeSetFinishedCompact&UnsafeReadFinishedmpact work well together.
```
Comment (ptabor) [inline server/mvcc/store_test.go:57]: FTR UnsafeReadFinishedCompat

File: server/mvcc/store_test.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	"go.etcd.io/etcd/server/v3/mvcc/backend"
+	betesting "go.etcd.io/etcd/server/v3/mvcc/backend/testing"
+	"go.etcd.io/etcd/server/v3/mvcc/buckets"
+)
+
+// TestScheduledCompact ensures that UnsafeSetScheduledCompact&UnsafeReadScheduledCompact work well together.
+func TestScheduledCompact(t *testing.T) {
```
Comment (ptabor) [inline server/mvcc/store_test.go:17]: Thank you for adding the test.

**Review by ptabor — APPROVED** [review]

File: server/mvcc/kvstore.go (hunk 1 above)
Comment (serathius) [reply server/mvcc/kvstore.go]: Done.

File: server/mvcc/store.go (hunk 2 above)
Comment (serathius) [reply server/mvcc/store.go]: Done.

File: server/mvcc/store_test.go (hunk 4 above)
Comment (serathius) [reply server/mvcc/store_test.go:17]: np

File: server/mvcc/store_test.go (hunk 3 above)
Comment (serathius) [reply server/mvcc/store_test.go]: Done.

**Pull request merged**